### PR TITLE
Fixed #8280 -- Allowed management command discovery for eggs

### PR DESCRIPTION
**Pull request by claudep**
Thanks jdetaeye for the report, and bhuztez and jdetaeye for the
initial patches.

### CODE REVIEW

**Comment by claudep** [comment]
buildbot, please retest this

File: tests/user_commands/tests.py [hunk 1]
```diff
@@ -72,6 +74,17 @@ def test_find_command_without_PATH(self):
...
+    def test_discover_commands_in_eggs(self):
+        """
+        Test that management commands can also be loaded from Python eggs.
+        """
+        egg_dir = '%s/eggs' % os.path.dirname(upath(__file__))
+        egg_name = '%s/basic.egg' % egg_dir
+        with extend_sys_path(egg_name):
+            with self.settings(INSTALLED_APPS=['commandegg']):
+                cmds = find_commands(os.path.join(apps.get_app_config('commandegg').path, 'management'))
+        self.assertListEqual(cmds, ['eggcommand'])
```
Comment (berkerpeksag) [inline tests/user_commands/tests.py]: Nitpick: assertEqual calls assertListEqual if both a and b is a list: See https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L389 and https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L777

**Comment by claudep** [comment]
There is a bug somewhere on Python 3.2 with `pkgutil.iter_modules`. At some point, `sys.path_importer_cache` is populated with an `importlib._bootstrap._FileFinder` instance which always return an empty list for `iter_modules`. I'm a bit stuck right now.

File: tests/user_commands/tests.py (hunk 1 above)
Comment (claudep) [inline tests/user_commands/tests.py]: Thanks for the tip, good to know!

**Comment by claudep** [comment]
Seems the introduced workaround solved the issue with 3.2. I didn't feel like investigating more, considering that we should drop 3.2 support soon.

**Pull request merged**